### PR TITLE
Mobile table view

### LINKED ISSUE #7552: Mobile table view
Customize the appearance of the table in mobile view, to be complies with:
https://cxwiki.sap.com/display/spar/B2B+My+Company-+Users

### PR DESCRIPTION
**Pull request by Matejk00**
Closes: #7552

### CODE REVIEW

File: projects/storefrontlib/src/shared/components/table/table.component.html [hunk 1]
```diff
@@ -1,40 +1,138 @@
...
+  ></ng-container>
+  <ng-container
+    *ngTemplateOutlet="button; context: { row: data.row, column: data.column }"
+  ></ng-container>
+  <ng-container
+    *ngTemplateOutlet="link; context: { row: data.row, column: data.column }"
+  ></ng-container>
+  <ng-container
+    *ngTemplateOutlet="text; context: { row: data.row, column: data.column }"
+  ></ng-container>
```
Comment (Pucek9) [inline projects/storefrontlib/src/shared/components/table/table.component.html]:
```suggestion
  <ng-container *ngTemplateOutlet="checkbox; context: data"></ng-container>
  <ng-container *ngTemplateOutlet="button; context: data"></ng-container>
  <ng-container *ngTemplateOutlet="link; context: data"></ng-container>
  <ng-container *ngTemplateOutlet="text; context: data"></ng-container>
```

File: projects/storefrontlib/src/shared/components/table/table.component.ts [hunk 2]
```diff
@@ -1,31 +1,44 @@
...
   checkBox?: boolean;
+  button?: boolean;
 }
 
 @Component({
   selector: 'cx-table',
   templateUrl: './table.component.html',
 })
 export class TableComponent {
+  iconTypes = ICON_TYPE;
```
Comment (Pucek9) [inline projects/storefrontlib/src/shared/components/table/table.component.ts]: seems as unused

File: projects/storefrontlib/src/shared/components/table/table.component.html [hunk 3]
```diff
@@ -1,40 +1,138 @@
...
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+  <div class="mobile mode">
+    <div class="cx-row" *ngFor="let row of tableData">
+      <div
+        *ngFor="let column of columns"
+        [ngClass]="{ 'cx-action': column?.checkBox || column?.button }"
```
Comment (Pucek9) [inline projects/storefrontlib/src/shared/components/table/table.component.html]: don't see results and even benefits from that

File: projects/storefrontlib/src/shared/components/table/table.component.html [hunk 4]
```diff
@@ -1,40 +1,138 @@
...
+    />
+  </ng-container>
+</ng-template>
+
+<!-- 
+  Template for rendering icon button if `button` property exists in column data. 
+
+  Icon button can be used as follows in tableData object:
+  
+  key: {
```
Comment (Pucek9) [inline projects/storefrontlib/src/shared/components/table/table.component.html]: as I mention, it's not nice to keep metadata together with data, that properties setup could be done in html

File: projects/storefrontlib/src/shared/components/table/table.component.html (hunk 1 above)
Comment (Matejk00) [reply projects/storefrontlib/src/shared/components/table/table.component.html]: Replaced.

File: projects/storefrontlib/src/shared/components/table/table.component.html (hunk 3 above)
Comment (Matejk00) [reply projects/storefrontlib/src/shared/components/table/table.component.html]: Left from previous version, we don't need that for now.

File: projects/storefrontlib/src/shared/components/table/table.component.html (hunk 4 above)
Comment (Matejk00) [reply projects/storefrontlib/src/shared/components/table/table.component.html]: Done.

File: projects/storefrontlib/src/shared/components/table/table.component.ts (hunk 2 above)
Comment (Matejk00) [reply projects/storefrontlib/src/shared/components/table/table.component.ts]: True, we can delete this.

**Review by Pucek9 — APPROVED** [review]

**Pull request merged**